wcfsetup/install/files/lib/action/DeleteSessionAction.class.php

```diff
@@ -0,0 +1,70 @@
+<?php
+namespace wcf\action;
+use wcf\system\exception\IllegalLinkException;
+use wcf\system\session\SessionHandler;
+use wcf\system\WCF;
+use wcf\util\JSON;
+use wcf\util\StringUtil;
+
+/**
+ * Deletes a specific user session.
+ *
+ * @author	Joshua Ruesweg
+ * @copyright	2001-2020 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\Action
+ */
+class DeleteSessionAction extends AbstractAction {
+	/**
+	 * @inheritDoc
+	 */
+	public $loginRequired = true;
+	
+	/**
+	 * @var string
+	 */
+	public $sessionID;
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function readParameters() {
+		parent::readParameters();
+		
+		if (isset($_POST['sessionID'])) $this->sessionID = StringUtil::trim($_POST['sessionID']);
+		
+		if (empty($this->sessionID)) {
+			throw new IllegalLinkException();
+		}
+		
+		$found = false;
+		foreach (SessionHandler::getInstance()->getUserSessions(WCF::getUser()) as $session) {
+			if ($session->getSessionID() === $this->sessionID) {
+				$found = true;
+				break;
+			}
+		}
+		
+		if (!$found) {
+			throw new IllegalLinkException();
+		}
```

Comment:
I'm not sure whether this check is necessary. It suffers from a TOCTOU problem anyway.

Reply:
If the session ID no longer exists, the session handler ignores this and handles it correctly. Nevertheless I think that a short test is not wrong, even if it is finally superfluous. So errors are much faster to detect if the action

Finally, the test eats up no real resources and is performed with methods that we have already programmed anyway.